### ISSUE TITLE
CSV Maximum line length not always checked in 0.8.1

### ISSUE THREAD

**Issue by michaeljohnalbers** [duckdb/duckdb#8320]
### What happens?

While upgrading to 0.8.1 (from 0.7.1) I'm seeing some unexpected behavior with the maximum line length in CSV files. The default maximum line length is 2097152 bytes. With 0.8.1 I'm seeing some different behavior, namely around parallel reads. Different cases follow (all writing is done with UTF-8 encoding).

This might have to do with the fact we're reading from GZIP compressed CSVs.

**Single line too long**
`"a".repeat(2 * 1024 * 1024 + 10);`
With parallel reading off, DuckDB throws this exception as it did in 0.7.1.
With parallel reading on, no exception is thrown. I tried up to a 200MB string with no exception thrown

**Single line too long, but with actual newline at the end**
`"a".repeat(2 * 1024 * 1024 + 10) + "\n";`
No exception is thrown regardless of parallel settings.

**Multiple lines too long**
`String value = "a".repeat(2 * 1024 * 1024 + 10) + "\n"; String data = value + value + value + value;`
With parallel reading off, no exception is thrown.
With parallel reading on, I get an exception, but something very different
> CSV File not supported for multithreading. This can be a problematic line in your CSV File or that this CSV can't be read in Parallel. Please, inspect if the line 2 is correct. If so, please run single-threaded CSV Reading by setting parallel=false in the read_csv call.

### To Reproduce

```java
    public void testCase() throws Exception {
        String value = "a".repeat(2 * 1024 * 1024 + 10);

        System.out.println("Basic line too long");
        Path case1 = Paths.get("line_too_long.csv");
        try {
            writeCsv(case1, value);
            runCopy(case1);
        } finally {
            case1.toFile().delete();
        }

        System.out.println("Basic line too long, with newline");
        Path case2 = Paths.get("line_too_long_with_newline.csv");
        try {
            writeCsv(case2, value + "\n");
            runCopy(case2);
        } finally {
            case2.toFile().delete();
        }

        System.out.println("Multiple lines too long");
        Path case3 = Paths.get("multiple_line_too_long.csv");
        try {
            String line = value + "\n";
            writeCsv(case3, line + line + line + line);
            runCopy(case3);
        } finally {
            case3.toFile().delete();
        }
    }

    public static void writeCsv(Path file, String data) throws IOException {
        try (GZIPOutputStream gzipOutputStream = new GZIPOutputStream(Files.newOutputStream(file))) {
            gzipOutputStream.write(data.getBytes(StandardCharsets.UTF_8));
        }
    }

    public static void runCopy(Path csv) throws Exception {
        List<Boolean> parallel = List.of(false, true);
        try (DuckDBConnection connection = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:")) {
            try (Statement s = connection.createStatement()) {
                s.execute("CREATE TABLE T1 (name VARCHAR)");
            }
            for (boolean p : parallel) {
                System.out.println("  Parallel = " + p);
                try (Statement s = connection.createStatement()) {
                    s.execute("COPY T1(name) from '" + csv + "' (DELIMITER ',', HEADER, COMPRESSION gzip, ALLOW_QUOTED_NULLS false, PARALLEL " + p + ")");
                    System.out.println("    Successful copy.");
                } catch (SQLException e) {
                    System.out.println("    Got Exception: " + e.getMessage());
                }
            }
        }
    }
```

Output
```
Basic line too long
  Parallel = false
    Got Exception: Invalid Input Error: Maximum line size of 2097152 bytes exceeded on line 1!
  Parallel = true
    Successful copy.
Basic line too long, with newline
  Parallel = false
    Successful copy.
  Parallel = true
    Successful copy.
Multiple lines too long
  Parallel = false
    Successful copy.
  Parallel = true
    Got Exception: Invalid Input Error: CSV File not supported for multithreading. This can be a problematic line in your CSV File or that this CSV can't be read in Parallel. Please, inspect if the line 2 is correct. If so, please run single-threaded CSV Reading by setting parallel=false in the read_csv call.
```

### OS:

MacOS

### DuckDB Version:

0.8.1

### DuckDB Client:

JDBC

### Full Name:

Michael Albers

### Affiliation:

ThoughtSpot (formerly Mode Analytics)

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by michaeljohnalbers** [comment]
@pdet Do you have any information about this? What happens with longer lines? Are they still inserted? I'm trying to determine if we can use 0.8.1 (especially with parallel CSV reading) in a production environment.

**Comment by pdet** [comment]
> @pdet Do you have any information about this? What happens with longer lines? Are they still inserted? I'm trying to determine if we can use 0.8.1 (especially with parallel CSV reading) in a production environment.

Hi @michaeljohnalbers, I'm currently refactoring the whole of the CSV Sniffer and Parser (See #7213 as a reference). In the PR I have open that rewrites a lot of the sniffer (See #8253) I've also added tests for this issue, so it should be fixed on the bleeding-edge version after  #8253 gets merged.

**Comment by szarnyasg** [comment]
@pdet I re-ran this code with the latest nightly build of the DuckDB JAR: `duckdb_jdbc-0.9.0-20230907.015504-331.jar`.
Java project here: https://github.com/szarnyasg/csv-reprod

```text
Basic line too long
  Parallel = false
    Got Exception: Invalid Input Error: Maximum line size of 2097152 bytes exceeded on line 1!
  Parallel = true
    Successful copy.
Basic line too long, with newline
  Parallel = false
    Got Exception: Invalid Input Error: Error in file "line_too_long_with_newline.csv" on line 1: Maximum line size of 2097152 bytes exceeded!
  Parallel = true
    Successful copy.
Multiple lines too long
  Parallel = false
    Got Exception: Invalid Input Error: Error in file "multiple_line_too_long.csv" on line 1: Maximum line size of 2097152 bytes exceeded!
  Parallel = true
    Got Exception: Invalid Input Error: CSV File not supported for multithreading. This can be a problematic line in your CSV File or that this CSV can't be read in Parallel. Please, inspect if the line 2 is correct. If so, please run single-threaded CSV Reading by setting parallel=false in the read_csv call.
```
